File: packages/go/generate_territorial_summary.go
package perspatial

import (
	"strconv"
	"strings"
)

func GenerateTerritorialSummary(intersections []Intersection) string {
	if len(intersections) == 0 {
		return "El territorio no presenta superposiciones con las capas evaluadas."
	}
	dominant := "BAJO"
	entities := make([]string, 0, len(intersections))
	for _, in := range intersections {
		if RiskLevelToScore(in.Level) > RiskLevelToScore(dominant) {
			dominant = in.Level
		}
		entities = append(entities, in.Entity)
	}
	n := strconv.Itoa(len(intersections))
	return "El territorio presenta superposicion de nivel " + dominant +
		" con " + n + " entidad(es): " + strings.Join(entities, ", ") + "."
}

File: packages/java/src/com/peruspatial/territorial/GenerateTerritorialSummary.java
package com.peruspatial.territorial;

import java.util.List;
import java.util.stream.Collectors;

public final class GenerateTerritorialSummary {
  private GenerateTerritorialSummary() {}

  public static String generateTerritorialSummary(List<Intersection> intersections) {
    if (intersections == null || intersections.isEmpty()) {
      return "El territorio no presenta superposiciones con las capas evaluadas.";
    }
    String dominant = "BAJO";
    for (Intersection i : intersections) {
      if (rank(i.level()) > rank(dominant)) dominant = i.level();
    }
    String entities = intersections.stream()
        .map(Intersection::entity)
        .collect(Collectors.joining(", "));
    return "El territorio presenta superposicion de nivel " + dominant + " con "
        + intersections.size() + " entidad(es): " + entities + ".";
  }

  private static int rank(String level) {
    if (level == null) return 0;
    return switch (level) {
      case "BAJO" -> 1;
      case "MEDIO" -> 2;
      case "ALTO" -> 3;
      default -> 0;
    };
  }
}
